server/service/src/dashboard/item_count.rs

```diff
@@ -0,0 +1,278 @@
+use std::convert::TryInto;
+
+use repository::{ItemFilter, ItemRepository, RepositoryError};
+
+use crate::{item_stats::get_item_stats, service_provider::ServiceContext};
+
+pub struct ItemCounts {
+    pub total: i64,
+    pub no_stock: i64,
+    pub low_stock: i64,
+}
+
+pub trait ItemCountServiceTrait: Send + Sync {
+    /// # Arguments
+    ///
+    /// * i32 threshold number of months below which is considered low stock
+    fn get_item_counts(
+        &self,
+        ctx: &ServiceContext,
+        store_id: &str,
+        low_stock_threshold: i32,
+    ) -> Result<ItemCounts, RepositoryError> {
+        ItemServiceCount {}.get_item_counts(ctx, store_id, low_stock_threshold)
+    }
+}
+
+pub struct ItemServiceCount {}
+
+impl ItemCountServiceTrait for ItemServiceCount {
+    fn get_item_counts(
```

Comment:
I think we should just use ItemStats here, don't see the need for count_no_stock: https://github.com/openmsupply/open-msupply/compare/927-item-stats...927-Example-use-item-state